x-pack/plugins/security_solution/cypress/data/detection_engine.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  From,
+  RiskScore,
+  Severity,
+  Threat,
+  ThreatSubtechnique,
+  ThreatTechnique,
+} from '@kbn/securitysolution-io-ts-alerting-types';
+import type {
+  Description,
+  FalsePositives,
+  Index,
+  Interval,
+  Name,
+  Note,
+  Query,
+  References,
+  Tags,
+} from '../../common/detection_engine/schemas/common';
```

Comment:
Regarding reusing already defined types: types imported here are  defined for the values that are used on API level. I.e. ones that are used when rules are created/updated through API calls.

But the functions defined here are used for filling control values in form, which can not be the same type
This leads to a situation like this, where risk scores have to be converted to a string type, which can be consumed by the cypress method.

```typescript
export const fillRiskScore = (riskScore: string = getRiskScore().toString()) => {
  cy.get(DEFAULT_RISK_SCORE_INPUT).type(`{selectall}${riskScore}`, { force: true });
};
```

It might be ok though, just something that caught my eye